src/libCli/ConnectionManager.hpp

```diff
@@ -14,137 +14,225 @@
+                channelCreds = generateSSLCredentials(sslClientCertPath, sslClientKeyPath, sslServerCertPath);
+                connection.channel = grpc::CreateChannel(f_serverAddress, channelCreds);
+            }
+            else
             {
-                ConnList connection;
+                // create insecure channel by default
                 connection.channel = grpc::CreateChannel(f_serverAddress, grpc::InsecureChannelCredentials());
-                connection.descDb = std::make_shared<grpc::ProtoReflectionDescriptorDatabase>(connection.channel);
-                connection.descPool = std::make_shared<grpc::protobuf::DescriptorPool>(connection.descDb.get());
-                connections[f_serverAddress] = connection;
             }
+
+            //std::cout << "Created Channel: " << connection.channel << std::endl;
+            connection.descDb = std::make_shared<grpc::ProtoReflectionDescriptorDatabase>(connection.channel);
+            connection.descPool = std::make_shared<grpc::protobuf::DescriptorPool>(connection.descDb.get());
+            connections[f_serverAddress] = connection;
+        }
+
+        /// @param f_sslClientCertPath Path to client certificate. Provide certificate as .pem or .crt
+        /// @param f_sslClientKeyPath Path to client private key. Provide key as .pem or .key
+        /// @param f_sslServerCretPath Path to server certificate. Provide Certificate as .pem or .crt
+        /// @return std::shared_ptr<grpc::ChannelCredentials> gRPC credentials as used for creating an SSL/TLS channel
+        std::shared_ptr<grpc::ChannelCredentials> generateSSLCredentials(const std::string f_sslClientCertPath, const std::string f_sslClientKeyPath, const std::string f_sslServerCertPath)
+        {
+            std::string clientKey = gwhisper::util::readFromFile(f_sslClientKeyPath);
+            if (clientKey == "FAIL")
+            {
+                std::cerr << "Error while fetching clientKey from: " << f_sslClientKeyPath << std::endl;
+                std::cerr << "Failed to build secure channel" << std::endl;
+                exit(EXIT_FAILURE);
+            }
+
+            std::string clientCert = gwhisper::util::readFromFile(f_sslClientCertPath);
+            if (clientCert == "FAIL")
+            {
+                std::cerr << "Error while fetching clientCert from: " << f_sslClientCertPath << std::endl;
+                std::cerr << "Failed to build secure channel" << std::endl;
+                exit(EXIT_FAILURE);
+            }
+
+            std::string serverCert = gwhisper::util::readFromFile(f_sslServerCertPath);
+            if (serverCert == "FAIL")
+            {
+                std::cerr << "Error while fetching serverCert from: " << f_sslServerCertPath << std::endl;
+                std::cerr << "Failed to build secure channel" << std::endl;
+                exit(EXIT_FAILURE);
+            }
+
+            grpc::SslCredentialsOptions sslOpts;
+
+            if (f_sslClientKeyPath != "")
```

Comment:
how can we go into this `if()`? I think in the case when `f_sslClientKeyPath == ""` is true file read would already had error and returned failure which causes program to terminate. Maybe move this up and check for empty string before reading the file?
Do we have a test for this (--ssl without cert paths?)